addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,22 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already registered with another partner."""
+        for bank in self:
+            duplicate = self.env['res.partner.bank'].search([('acc_number', '=', bank.acc_number)], limit=1)
+            bank.duplicate_account_warning = False
+            if not bank.acc_number or not duplicate:
+                continue
+            if bank.id != duplicate.id:
+                bank.duplicate_account_warning = _(
+                    "Warning : The Bank Account could be a duplicate of %(name)s (<a href=%(url)s>%(partner)s</a>)",
```

Comment:
Since we handle multiple bank accounts, we should probably put the account number in the message as well.